WWCodeTokyo/speak-her-db
# feat: adds self-nomination checkbox

References:
https://github.com/WWCodeTokyo/speak-her-db/issues/163

**Proposed Changes**
Adds a switch that can be toggled to self nominate. This switch will automatically fill in the previously filled in email and english name. On top of that, if the toggle is active, any input to the english name and email is automatically propagated to the submitter fields as well.

![image](https://user-images.githubusercontent.com/18051009/94937448-8d197600-050a-11eb-96e5-e6da0befca4f.png)


*Description of changes in this PR*
- Adds a new row on top of the submitter fields which holds a simple switch from vuetify.
- Handles the switch to auto-propagate the submitter fields both on toggle as well as on input of email as well as the english name.
- Handles the off toggle of the switch to reset the values again.

**Testing Plan**
I didn't see a test setup yet so I kept the PR without any tests. Let me know if you want me to write any unit or GUI tests.

*How do the reviewers QA this? Feel free to write unit tests or describe manual testing steps here*
Case 1:
- Go to the `Nominate a speaker` page
- Fill in the english name and email
- Toggle the switch
- Submitter fields should be auto propagated
- Make changes to the english name and email fields
- Changes should have been auto-propagated to the submitter fields again

Case 2:
- Go to the `Nominate a speaker` page
- Don't fill in anything
- Toggle the switch
- Fill in the english name and email
- Input of english name and email should be auto-propagated to the submitter fields

Case 3:
- Go to the `Nominate a speaker` page
- Fill in the name and email
- Toggle the switch
- Toggle the switch again
- Submitter fields should be reset
